src/protagonist/Engine/Infrastructure/ServiceCollectionX.cs

```diff
@@ -118,6 +122,8 @@ public static IServiceCollection AddAssetIngestion(this IServiceCollection servi
     public static IServiceCollection AddDataAccess(this IServiceCollection services, IConfiguration configuration)
         => services
             .AddScoped<IPolicyRepository, PolicyRepository>()
+            .AddScoped<IAssetRepository, AssetRepository>()
+            .AddScoped<IEntityCounterRepository, EntityCounterRepository>()

```

Comment:
Engine shouldn't require `IEntityCounterRepository` as it never touches entity-counters.

Reply:
it's used by `AssetRepository` to decrement `SpaceImages` and `CustomerImages` on delete - looking at this a bit deeper, it looks like there's a method in `EngineAssetRepository` which does something similar to what I need, so will look to remove `AssetRepository`

Reply:
so I've managed to do this, but probably needs a discussion as there's a small change that has to be made to the `EngineAssetRepository` where it changes a `bool` that I'm not sure on the impact of